app/src/main/java/com/nervousfish/nervousfish/exceptions/NoBluetoothException.java

```diff
@@ -0,0 +1,24 @@
+package com.nervousfish.nervousfish.exceptions;
+
+/**
+ * Simple {@link RuntimeException} that is used to notify usage of Bluetooth is not avaiable.
+ */
+public class NoBluetoothException extends RuntimeException {
```

Comment:
class metrics, because you can't run SIG locally. instance field access / methods not qualified with this because you like to use this apparently. Everything from concurrency because it really are good practices and Checkstyle, PMD and Findbugs don't know anything about those. Everything from serialization, because serialversionuid: (copied from the documentation)

> The serialization runtime associates with each serializable class a version number, called a serialVersionUID, which is used during deserialization to verify that the sender and receiver of a serialized object have loaded classes for that object that are compatible with respect to serialization. If the receiver has loaded a class for the object that has a different serialVersionUID than that of the corresponding sender's class, then deserialization will result in an  InvalidClassException. If a serializable class does not explicitly declare a serialVersionUID, then the serialization runtime will calculate a default serialVersionUID value for that class based on various aspects of the class, as described in the Java(TM) Object Serialization Specification. **However, it is strongly recommended** that all serializable classes explicitly declare serialVersionUID values, since the default serialVersionUID computation is highly sensitive to class details that may vary depending on compiler implementations, and can thus result in unexpected InvalidClassExceptions during deserialization. Therefore, to guarantee a consistent serialVersionUID value across different java compiler implementations, a serializable class must declare an explicit serialVersionUID value. 

Also, http://index-of.es/JAVA/Effective%20Java.pdf , very bottom page 301. The readobject/writeobject are needed to that we can extend the exception with new fields in a later release. Great explanation at the beginning of page 299, http://index-of.es/JAVA/Effective%20Java.pdf